jbpm-workitems/jbpm-workitems-rest/src/test/java/org/jbpm/process/workitem/rest/RestWorkItemHandlerTest.java

```diff
@@ -1253,4 +1254,135 @@ public void testHandleErrorOnNotSuccessfulResponseHandlingException() {
+		String result = (String) results.get(PARAM_RESULT);
+		assertEquals(cookie, result);
+	}
+	
+	@Test
+	public void testCookieSetCookiePath() {
+		RESTWorkItemHandler handler = new RESTWorkItemHandler();
+		String cookieName1 = "cookieParam1";
+		String cookieValue1 = "cookieParam1_Value";
+		String cookie = cookieName1 + "=" + cookieValue1;
+
+		WorkItemImpl workItem = new WorkItemImpl();
+		workItem.setParameter("Url",
+				      serverURL + "/testSingleCookie");
+		workItem.setParameter("Method",
+				      "GET");
+		workItem.setParameter("Cookie",
+				      cookie);
+		workItem.setParameter("CookiePath",
+				      "/test/testSingleCookie");
+
+		WorkItemManager manager = new TestWorkItemManager();
+		handler.executeWorkItem(workItem, manager);
+		Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+		String result = (String) results.get(PARAM_RESULT);
+		assertEquals(cookie, result);
+	}
+
+	@Test
+	public void testCookieIncorrectSeparators() {
+		RESTWorkItemHandler handler = new RESTWorkItemHandler();
+		String cookieName1 = "cookieParam1";
+		String cookieValue1 = "Value1,Value2";
+		String cookieName2 = "cookieParam2";
+		String cookieValue2 = "Value1,Value2";
+
+		String cookie = cookieName1 + "=" + cookieValue1 + ":" + cookieName2 + "=" + cookieValue2;
+
+		WorkItemImpl workItem = new WorkItemImpl();
+		workItem.setParameter("Url",
+				      serverURL + "/testSingleCookie");
+		workItem.setParameter("Method",
+				      "GET");
+		workItem.setParameter("Cookie",
+				      cookie);
+
+		WorkItemManager manager = new TestWorkItemManager();
+		handler.executeWorkItem(workItem, manager);
+		Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+		String result = (String) results.get(PARAM_RESULT);
+		assertEquals(cookie, result);
+	}
+
+	@Test
+	public void testCookieMultipleCookie() {
+		RESTWorkItemHandler handler = new RESTWorkItemHandler();
+		String cookieName1 = "cookieParam1";
+		String cookieValue1 = "cookieParam1_Value";
+		String cookieName2 = "cookieParam2";
+		String cookieValue2 = "cookieParam2_Value";
+		String cookie = cookieName1 + "=" + cookieValue1 + ";" + cookieName2 + "=" + cookieValue2;
+
+		WorkItemImpl workItem = new WorkItemImpl();
+		workItem.setParameter("Url",
+				      serverURL + "/testMultipleCookie");
+		workItem.setParameter("Method",
+				      "GET");
+		workItem.setParameter("Cookie",
+				      cookie);
+
+		WorkItemManager manager = new TestWorkItemManager();
+		handler.executeWorkItem(workItem, manager);
+		Map<String, Object> results = ((TestWorkItemManager) manager).getResults(workItem.getId());
+		String result = (String) results.get(PARAM_RESULT);
+		assertEquals(cookie, result);
+	}
+
+	@Test
+	public void testCookieIncorrectCookie() {
+		RESTWorkItemHandler handler = new RESTWorkItemHandler();
+		String cookieName1 = "cookieParam1";
+		String cookieValue1 = null;
```

Comment:
Same as explained before: java will convert it to "null" string.
Perhaps you could test Cookie like "key=value;=" or without separator ";"

Reply:
Test case modified as per suggestion.